cli/flox-rust-sdk/src/data/mod.rs

```diff
@@ -1,10 +1,6 @@
-mod canonical_path;
-mod version;
-
 use std::fmt::Display;
 
-pub use canonical_path::{CanonicalPath, CanonicalizeError};
-pub use version::Version;
+pub use shared::canonical_path::{CanonicalPath, CanonicalizeError};
```

Comment:
If we like having a `shared` crate I can get rid of the re-export and update all use statements throughout, but I figured we should confirm we want to add the `shared` crate first

Reply:
I don't _love_ it because of the extra scaffolding and design rules. I guess the alternatives are:
- duplicating those core functions: which I love even less in case they diverge
- pulling in `flox-rust-sdk`: there's a lot of [discussion in the doc](https://docs.google.com/document/d/1XLpxgJx-x_yBFCQVyRy6QKDMa27UkrB27QssMXhqL7M/edit?pli=1#heading=h.nedarqgctpz2) about how to keep the `flox-activations` binary small but not much about _why_ it's so important?

Reply:
`flox-activations` will end up in any container we produce. `flox-rust-sdk` depends on `nix`, which has a closure of about ~100 MB. Which would mean we'd take a 100 MB penalty for every one of our containers

Reply:
yeah i we shouldnt depend on all of flox-rust-sdk
we can potentially add feature flags for some reexports of that crate though if we really want to avoid a shared crate.

Generally i dont see too big of an issue with splitting out shared utils

Reply:
I'm not sure, but it seems like feature flagging would be just as annoying if not worse

Reply:
Feature flagging would be annoying for sure. I think a shared crate (not called `shared`) is the way to go for functions and data that need to be used across all of our utilities. At this point we have a small collection of different tools that need some of the same data and operations, but don't necessarily need to depend on each other in their entirety, and we should start factoring out exactly what is needed by what.

Reply:
Basically I would call the shared crate `flox-core`, and if we really wanted to make me happy rename `flox-rust-sdk` -> `flox-lib`